### PR TITLE
🧪 Add edge case tests for parsePositiveInt

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:**
Added additional tests for `parsePositiveInt` utility function in `@paper-tools/core` to increase test reliability. Addressed the edge cases with strings that have spaces or trailing characters, as well as testing math-related strings like `Infinity` and `NaN`.

📊 **Coverage:**
The tests now handle strings with leading/trailing spaces (`"  42  "`), trailing non-numeric characters (`"42abc"`), and invalid infinite limits like `"Infinity"`, `"-Infinity"` and `"NaN"`.

✨ **Result:**
The test suite is more robust when dealing with input that shouldn't be fully trusted, covering mathematical boundaries. Test coverage remains at 100%.

---
*PR created automatically by Jules for task [4951439846594450550](https://jules.google.com/task/4951439846594450550) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

前回レビューの指摘（`"42abc"` 受け入れ問題・非10進記法の抜け）に対応した実装改善とテスト追加 PR です。`parsePositiveInt` の実装を `Number.parseInt` から `value.trim()` + `/^[0-9]+$/` 正規表現 + `Number()` 方式に変更し、厳密なバリデーションを実現しています。

- **実装変更**: `value.trim()` による空白トリム、`/^[0-9]+$/` 正規表現で純粋な10進正整数文字列のみを許可し、`Infinity`・`NaN`・`0x10`・`1e2`・`42abc` などをすべてエラーとして弾く。
- **テスト追加**: 空白トリム・末尾文字・`Infinity`/`NaN`・非10進記法（16進・2進・8進・科学表記）の各ケースが独立した `it` ブロックで網羅されており、前回指摘事項がすべて解消されている。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

テスト追加と実装の小規模な厳格化のみで、既存の正常系動作に変更はなく安全にマージできます。

変更は parsePositiveInt の入力バリデーション強化（/^[0-9]+$/ 正規表現の導入と trim() 追加）と、それに対応するテストの追加のみです。正常系（純粋な正の整数文字列）の動作は変わらず、エラー系が追加・厳格化されているため、既存の利用箇所に悪影響を与えるリスクはほぼありません。前回レビューで指摘されたすべての問題が適切に対処されています。

特に注意が必要なファイルはありません。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/core/src/utils.ts | parsePositiveInt の実装を強化。value.trim() + /^[0-9]+$/ 正規表現 + Number() による変換に変更し、16進数・科学表記・末尾文字・Infinity を正しく弾くようになった。 |
| packages/core/tests/utils.test.ts | 前回レビューで指摘されたケース（"42abc" をエラー化、非10進記法テスト追加）が適切に対処され、空白トリム・Infinity/NaN・非10進記法の各テストも独立した it ブロックで整理されている。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["parsePositiveInt(value)"] --> B["normalizedValue = value.trim()"]
    B --> C{"/^[0-9]+$/.test(normalizedValue)?"}
    C -- "false\n('42abc', '0x10', 'NaN', '-1', ...)" --> E["throw Error\n'正の整数を指定してください: {value}'"]
    C -- true --> D["parsed = Number(normalizedValue)"]
    D --> F{"Number.isInteger(parsed)?"}
    F -- "false\n(Infinity など)" --> E
    F -- true --> G{"parsed <= 0?"}
    G -- "true ('0')" --> E
    G -- false --> H["return parsed ✅"]
```

<sub>Reviews (3): Last reviewed commit: ["🧪 Strict integer validation for parsePo..."](https://github.com/hiroki-org/paper-tools/commit/bbeedf0f7811b6f832d0ce3ddd21bc7d827b89b1) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32477136)</sub>

### CODE REVIEW

**Comment by is0692vs** [comment]
@greptile review

File: packages/core/tests/utils.test.ts [hunk 1]
```diff
@@ -35,6 +35,17 @@ describe("parsePositiveInt", () => {
         expect(() => parsePositiveInt("-5", null)).toThrow("正の整数を指定してください: -5");
         expect(() => parsePositiveInt("foo", undefined)).toThrow("正の整数を指定してください: foo");
     });
+
+    it("should handle strings with leading/trailing spaces and other characters", () => {
+        expect(parsePositiveInt("  42  ")).toBe(42);
+        expect(parsePositiveInt("42abc")).toBe(42);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/core/tests/utils.test.ts]: Addressed by making parsePositiveInt strict with Number(value) + Number.isInteger, so 42abc and decimal values now throw instead of being truncated. Verification: pnpm --filter @paper-tools/core test and build both pass.

[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/core/tests/utils.test.ts]: Addressed by separating the whitespace case from trailing-character validation, and by changing 42abc to an error expectation. The implementation now rejects non-integer numeric strings consistently.

**Comment by is0692vs** [comment]
@greptile review

File: packages/core/src/utils.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@
  * @throws 値が正の整数でない場合は Error をスロー
  */
 export function parsePositiveInt(value: string, optionName?: unknown): number {
-    const parsed = Number.parseInt(value, 10);
-    if (!Number.isFinite(parsed) || Number.isNaN(parsed) || parsed <= 0) {
+    const parsed = Number(value);
+    if (!Number.isInteger(parsed) || parsed <= 0) {
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/core/src/utils.ts]: Addressed by constraining parsePositiveInt to a trimmed decimal digit string before converting to Number, and by adding explicit rejection tests for 0x10, 0b101, 0o10, and 1e2. Verification: pnpm --filter @paper-tools/core test and build both pass.

File: packages/core/tests/openalex-client.test.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { beforeEach, describe, expect, it, vi } from "vitest";
 const mockFetch = vi.fn();
 vi.stubGlobal("fetch", mockFetch);
 
-import { resolveOpenAlexAuthorId } from "../src/openalex-client.js";
+import { getOpenAlexAuthor, resolveOpenAlexAuthorId } from "../src/openalex-client.js";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/core/tests/openalex-client.test.ts]: Addressed by removing the stray getOpenAlexAuthor import from the PR diff. I also reverted the unrelated test deletions and middleware/BibTeX/Crossref edits so this PR now only changes parsePositiveInt and its tests.
